tests/command.dev.test.js

```diff
@@ -122,7 +122,7 @@ testMatrix.forEach(({ args }) => {
     })
   })
 
-  test(testName('should override process env vars with ones in .env.development', args), async t => {
+  test(testName('should override .env.development with process env', args), async t => {
     await withSiteBuilder('site-with-override', async builder => {
```

Comment:
Looks like the initial intention of this feature was to just override vars set in the UI/netlify.toml:
https://github.com/netlify/cli/pull/736